dbt/include/bigquery/macros/utils/get_columns_spec_ddl.sql

```diff
@@ -1,18 +1,30 @@
 {% macro bigquery__get_columns_spec_ddl() %}
   {# loop through user_provided_columns to create DDL with data types and constraints #}
-    {%- set ns = namespace(at_least_one_check=False) -%}
+    {%- set ns = namespace(at_least_one_check=False, at_least_one_pk=False) -%}
     {%- set user_provided_columns = model['columns'] -%}
     (
-    {% for i in user_provided_columns -%}
+    {% for i in user_provided_columns %}
       {%- set col = user_provided_columns[i] -%}
-      {% set constraints = col['constraints'] -%}
-      {%- set ns.at_least_one_check = ns.at_least_one_check or col['constraints_check'] %}
-      {{ col['name'] }} {{ col['data_type'] }} {% for x in constraints %} {{ x or "" }} {% endfor %} {{ "," if not loop.last }}
-    {%- endfor %}
-  )
+      {%- set constraints = col['constraints'] -%}
+      {{ col['name'] }} {{ col['data_type'] }}
+      {%- for x in constraints -%}
+        {%- if x.type == "check" -%}
+          {%- set ns.at_least_one_check = True -%}
+        {%- elif x.type == "primary_key" -%}
+          {%- set ns.at_least_one_pk = True -%}
```

Comment:
do we need to add this check? GCP announced in the [22/02/2023 Google Cloud release notes](https://cloud.google.com/release-notes#February_22_2023) the beta of primary & foreign key table constraints - https://github.com/dbt-labs/dbt-bigquery/issues/568#issuecomment-1448836093

Reply:
Actually - it looks like there's still some limitation on how primary & foreign key constraints get specified (require a `not enforced` statement) . I'm also not sure how broadly this feature is turned on as a beta. Probably makes sense to leave your changes as-is and revisit it as part of https://github.com/dbt-labs/dbt-bigquery/issues/568